generic-templates/newsletter-signup-snippet/snippet.html

```diff
@@ -172,327 +198,214 @@
+
+          {% if reveal == "click" %}
+            var clickToReveal = document.querySelector(".click-to-reveal");
+            clickToReveal.addEventListener("click", function() {
+              this.classList.add("hide");
+              formContainer.classList.add("reveal");
               emailInput.focus();
+            });
+          {% endif %}
+
+          function submit() {
+            if (button.classList.contains("loading")) {
               return;
             }
+            changeState(000);
+            sendEmailToBasket();
           }
-          currentIndex++;
-          transitionTrack.style.transform = "translate(-" + ( currentIndex * trackItemWidth ) + "px, 0px)";
-        }
-        function onBack(e) {
-          if (currentIndex <= 0) {
-            return;
+
+          document.querySelector("#form").addEventListener("submit", function(e) {
+            e.preventDefault();
+            submit();
+          });
+
+          function sendEmailToBasket() {
+            var lang = "{{ locale|default('en-US', true) }}";
+            var url = "{{ basket_host|default('https://basket.mozilla.org', true) }}" + "/news/subscribe/";
+            var newsletters = "{{ newsletters }}";
+            var source_url = "https://snippets.mozilla.com/show/{{ snippet_id }}";
+            source_url = encodeURIComponent(source_url);
+            var email = emailInput.value;
+            var params = "newsletters=" + newsletters + "&source_url=" + source_url + "&lang=" + lang + "&email=" + email + "&trigger_welcome=N";
+            var request = new XMLHttpRequest();
+            request.onreadystatechange = function() {
+              if (request.readyState == 4) {
+                changeState(request.status);
+              }
+            }
+
+            request.open("POST", url);
+            request.send(params);
           }
-          currentIndex--;
-          transitionTrack.style.transform = "translate(-" + ( currentIndex * trackItemWidth ) + "px, 0px)";
-          basketError.classList.add("hidden");
-          successMessage.classList.add("hidden");
-        }
-
-        for (var nextIndex = 0; nextIndex < nextButtons.length; nextIndex++) {
-          nextButtons[nextIndex].addEventListener("click", onNext);
-        }
-
-        for (var backIndex = 0; backIndex < backButtons.length; backIndex++) {
-          backButtons[backIndex].addEventListener("click", onBack);
-        }
-
-        var elHighlighted = snippet.querySelector("em");
-        // Fade in highlight on text
-        setTimeout(function() {
-          if (elHighlighted) {
-            elHighlighted.classList.add("active");
+
+          function changeState(status) {
+            if (status === 200) {
+              button.classList.remove("loading");
+              basketError.classList.add("hidden");
+              successMessage.classList.remove("hidden");
+              formContainer.classList.add("hidden");
+              sendMetric("signup-success");
```

Comment:
Do you want to add to blocklist as well to not show the snippet to the user again?